Meta/run.py

```diff
@@ -0,0 +1,978 @@
+    ram_size: str = "1G"
+    disk_image: str = "_disk_image"
+    qemu_cpu: str = "max"
+    cpu_count: int = 2
+    extra_arguments: list[str] = field(default_factory=list)
+    machine_type: MachineType = MachineType.Default
+    spice_arguments: list[str] = field(default_factory=list)
+    audio_backend: str = "pa,timer-period=2000"
+    audio_device: list[str] = field(default_factory=list)
+    screen_count: int = 1
+    display_backend: str | None = None
+    display_device: str = "VGA,vgamem_mb=64"
+    boot_drive: list[str] = field(default_factory=list)
+    host_ip: str = "127.0.0.1"
+    ethernet_device_type: str = "e1000"
+    network_backend: str | None = None
+    network_default_device: list[str] = field(default_factory=list)
+    kernel_and_initrd: list[str] = field(default_factory=list)
+
+
+def kvm_usable() -> bool:
+    return access("/dev/kvm", os.R_OK | os.W_OK | os.X_OK)
+
+
+def determine_qemu_kind() -> QEMUKind:
+    if run(["command", "-v", "wslpath"], shell=True).returncode == 0:
+        # Assume native Windows QEMU for now,
+        # we might discard that assuption later when we properly
+        # look for the binary.
+        return QEMUKind.NativeWindows
+    if sys.platform == "darwin":
+        return QEMUKind.MacOS
+    if os.uname().sysname == "SerenityOS":
+        return QEMUKind.SerenityOS
+    return QEMUKind.Other
+
+
+def determine_serenity_arch() -> Arch:
+    match environ.get("SERENITY_ARCH"):
+        case "aarch64":
+            return Arch.Aarch64
+        case "x86_64":
+            return Arch.x86_64
+        case _:
+            raise RunError("Please specify a valid SerenityOS architecture")
+
+
+def determine_machine_type() -> MachineType:
+    provided_machine_type = environ.get("SERENITY_RUN")
+    if provided_machine_type is not None:
+        try:
+            value = MachineType(provided_machine_type)
+        except ValueError:
+            raise RunError(
+                f"{provided_machine_type} is not a valid SerenityOS machine type"
+            )
+        return value
+    return MachineType.Default
+
+
+def setup_qemu_binary(config: Configuration):
+    qemu_binary_basename: str | None = None
+    if "SERENITY_QEMU_BIN" in environ:
+        qemu_binary_basename = environ.get("SERENITY_QEMU_BIN")
+    else:
+        match config.architecture:
+            case Arch.Aarch64:
+                qemu_binary_basename = "qemu-system-aarch64"
+            case Arch.x86_64:
+                qemu_binary_basename = "qemu-system-x86_64"
+    if qemu_binary_basename is None:
+        raise RunError("QEMU binary could not be determined")
+
+    # Try finding native Windows QEMU first
+    if config.qemu_kind == QEMUKind.NativeWindows:
+        # Some Windows systems don't have reg.exe's directory on the PATH by default.
+        environ["PATH"] = environ["PATH"] + ":/mnt/c/Windows/System32"
+        qemu_install_dir_result = run(
```

Comment:
Let's grab this from the builtin winreg module instead
<https://docs.python.org/3/library/winreg.html>

... is what I would say if that module worked in wsl. There is a `wslwinreg` module though, but that needs a pip install :yakshrug:

Reply:
Even though this script makes Python 3.10 a hard requirement, requiring extra packages is still not nice since we don't need that anywhere else and don't have a venv setup.